File: backend/src/main/java/org/enactusensi/platform/entity/Project.java
package org.enactusensi.platform.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.Index;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OrderBy;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "projects", indexes = {
        @Index(name = "idx_project_slug", columnList = "slug", unique = true),
        @Index(name = "idx_project_display_order", columnList = "display_order")
})
public class Project extends BaseAuditableEntity {

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(nullable = false)
    private String name;

    @Column(name = "short_description", columnDefinition = "TEXT")
    private String shortDescription;

    @Column(name = "full_description", columnDefinition = "TEXT")
    private String fullDescription;

    @Column(columnDefinition = "TEXT")
    private String context;

    @Column(columnDefinition = "TEXT")
    private String solution;

    @Column(columnDefinition = "TEXT")
    private String impact;

    @Column(columnDefinition = "TEXT")
    private String objectives;

    @Column(name = "cover_image")
    private String coverImage;

    private String logo;

    private String category;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ProjectStatus status = ProjectStatus.IDEA;

    /** Enactus methodology stage, tracked for the project manager's space. */
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ProjectPhase phase = ProjectPhase.PROBLEMATIQUE;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(nullable = false)
    private boolean featured = false;

    @Column(name = "display_order", nullable = false)
    private Integer displayOrder = 0;

    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("displayOrder ASC")
    private List<ProjectImage> images = new ArrayList<>();
}

File: frontend/types/project.ts
export type ProjectStatus = "IDEA" | "IN_PROGRESS" | "ACTIVE" | "COMPLETED" | "ARCHIVED";
export type ProjectPhase = "PROBLEMATIQUE" | "IDEA_STEP_1" | "IDEA_STEP_2" | "PILOT_STEP_1" | "PILOT_STEP_2";

export interface ProjectImage {
  id: number;
  imageUrl: string;
  caption: string | null;
  displayOrder: number;
}

export interface ProjectSummary {
  id: number;
  slug: string;
  name: string;
  shortDescription: string | null;
  coverImage: string | null;
  logo: string | null;
  category: string | null;
  status: ProjectStatus;
  featured: boolean;
}

export interface ProjectDetail {
  id: number;
  slug: string;
  name: string;
  shortDescription: string | null;
  fullDescription: string | null;
  context: string | null;
  solution: string | null;
  impact: string | null;
  objectives: string | null;
  coverImage: string | null;
  logo: string | null;
  category: string | null;
  status: ProjectStatus;
  startDate: string | null;
  endDate: string | null;
  featured: boolean;
  images: ProjectImage[];
}

export interface AdminProject extends ProjectDetail {
  phase: ProjectPhase;
  displayOrder: number;
  createdAt: string;
  updatedAt: string;
}

export interface ProjectInput {
  name: string;
  shortDescription?: string;
  fullDescription?: string;
  context?: string;
  solution?: string;
  impact?: string;
  objectives?: string;
  coverImage?: string;
  logo?: string;
  category?: string;
  status: ProjectStatus;
  phase: ProjectPhase;
  startDate?: string | null;
  endDate?: string | null;
  featured: boolean;
  displayOrder?: number;
}
